### PR TITLE
Add Write implementation for Vec

### PR DESCRIPTION
**Pull request by jess-sol**
Allows for writing to a Vec which has a custom allocator. This allows crates which implement a Write interface but not allocator-api2 to still write to a Vec with a custom allocator.

### CODE REVIEW

**Review by zakarumych — CHANGES_REQUESTED** [review]
Thank you, this is copy of the standard `Write for Vec`, right?
In this case the only one detail missing is putting it under feature flag.

File: src/stable/vec/mod.rs [hunk 1]
```diff
@@ -67,6 +67,8 @@ use core::ops::{self, Bound, Index, IndexMut, Range, RangeBounds};
 use core::ptr::{self, NonNull};
 use core::slice::{self, SliceIndex};
 
+use std::io;
```
[the remark this answers is not in the document]
Comment (jess-sol) [reply src/stable/vec/mod.rs:71]: Oops missed that, added

**Comment by jess-sol** [comment]
Yep copied it from [latest stable](https://doc.rust-lang.org/stable/src/std/io/impls.rs.html#394-426) :+1:

**Comment by zakarumych** [comment]
Looks great! Thank you!

**Pull request merged**